harmony/http.py

```diff
@@ -302,21 +336,28 @@ def download(config, url: str, access_token: str, data, destination_file, user_a
     start_time = datetime.datetime.now()
     logger.info(f'timing.download.start {url}')
 
+    _preprocess_download_kwargs(download_kwargs, logger)
+
     if access_token is not None and _valid(config.oauth_host, config.oauth_client_id, access_token):
-        response = _download(config, url, access_token, data, user_agent)
+        response = _download(config, url, access_token, data, user_agent, stream=download_kwargs["stream"])
 
     if response is None or not response.ok:
         if config.fallback_authn_enabled:
             msg = ('No valid user access token in request or EDL OAuth authentication failed.'
                    'Fallback authentication enabled: retrying with Basic auth.')
             logger.warning(msg)
-            response = _download_with_fallback_authn(config, url, data, user_agent)
+            response = _download_with_fallback_authn(config, url, data, user_agent, stream=download_kwargs["stream"])
 
     if response.ok:
+        if not download_kwargs["stream"]:
+            destination_file.write(response.content)
+            file_size = sys.getsizeof(response.content)
+        else:
+            for chunk in response.iter_content(chunk_size=download_kwargs["chunk_size"]):
```

Comment:
Similar comments here. If we know there's a valid argument called "chunk_size" and we use it, it's probably best to spell it out in the method signature.

Also, though, this "chunk_size" doesn't mean what people familiar with HDF / Zarr / etc would assume it means, right?  I'd consider renaming it to avoid confusion.  Maybe part_size or block_size or ... I don't know :)

Reply:
buffer_size?